libraries/standard/http/src/private/http_client_internal.h

```diff
@@ -28,4 +28,84 @@
+#define NEWLINE_CHARACTER                  '\n'
+#define HTTP_HEADER_FIELD_SEPARATOR        ": "
+#define COLON_CHARACTER                    ':'
+#define SPACE_CHARACTER                    ' '
+#define EQUAL_CHARACTER                    '='
+#define DASH_CHARACTER                     '-'
+#define HTTP_HEADER_LINE_SEPARATOR_LEN     ( uint8_t ) ( sizeof( HTTP_HEADER_LINE_SEPARATOR ) - 1 )
+#define HTTP_HEADER_FIELD_SEPARATOR_LEN    ( uint8_t ) ( sizeof( HTTP_HEADER_FIELD_SEPARATOR ) - 1 )
+#define COLON_CHARACTER_LEN                ( uint8_t ) ( sizeof( COLON_CHARACTER ) - 1 )
+#define SPACE_CHARACTER_LEN                ( uint8_t ) ( sizeof( SPACE_CHARACTER ) - 1 )
+#define EQUAL_CHARACTER_LEN                ( uint8_t ) ( sizeof( EQUAL_CHARACTER ) - 1 )
+#define DASH_CHARACTER_LEN                 ( uint8_t ) ( sizeof( DASH_CHARACTER ) - 1 )
+
+/**
+ * @brief The maximum length for a 32-bit integer when converted to a string.
+ *
+ * This is used to initialize a local array for the final headers to send.
+ */
+#define INT32_STRING_MAX_LEN               ( uint8_t ) ( sizeof( INT32_STRING_MAX ) - 1 )
+
+
+/**
+ * @brief Constants for header fields added automatically during the request initialization.
+ */
+#define HTTP_USER_AGENT_FIELD                   "User-Agent"
+#define HTTP_HOST_FIELD                         "Host"
+#define HTTP_USER_AGENT_FIELD_LEN               ( uint8_t ) ( sizeof( HTTP_USER_AGENT_FIELD ) - 1 )
+#define HTTP_HOST_FIELD_LEN                     ( uint8_t ) ( sizeof( HTTP_HOST_FIELD ) - 1 )
+
+/**
+ * @brief Constants for header fields added based on flags.
+ */
+#define HTTP_CONNECTION_FIELD                   "Connection"
+#define HTTP_CONTENT_LENGTH_FIELD               "Content-Length"
+#define HTTP_CONNECTION_FIELD_LEN               ( uint8_t ) ( sizeof( HTTP_CONNECTION_FIELD ) - 1 )
+#define HTTP_CONTENT_LENGTH_FIELD_LEN           ( uint8_t ) ( sizeof( HTTP_CONTENT_LENGTH_FIELD ) - 1 )
+
+/**
+ * @brief Constants for header values added based on flags.
+ */
+#define HTTP_CONNECTION_KEEP_ALIVE_VALUE        "keep-alive"
+#define HTTP_CONNECTION_CLOSE_VALUE             "close"
+#define HTTP_CONNECTION_KEEP_ALIVE_VALUE_LEN    ( uint8_t ) ( sizeof( HTTP_CONNECTION_KEEP_ALIVE_VALUE ) - 1 )
+#define HTTP_CONNECTION_CLOSE_VALUE_LEN         ( uint8_t ) ( sizeof( HTTP_CONNECTION_CLOSE_VALUE ) - 1 )
+
+/**
+ * @brief Constants for header fields added for "Range" header.
+ */
+#define HTTP_RANGE_FIELD                        "Range"
+#define HTTP_RANGE_FIELD_LEN                    ( uint8_t ) ( sizeof( HTTP_RANGE_FIELD ) - 1 )
+
+/**
+ * @brief Constants for header value prefix added for "Range" header.
+ */
+#define HTTP_RANGE_BYTES_PREFIX_VALUE           "bytes"
+#define HTTP_RANGE_BYTES_PREFIX_VALUE_LEN       ( uint8_t ) ( sizeof( HTTP_RANGE_BYTES_PREFIX_VALUE ) - 1 )
+
+/**
+ * @brief Longest possible string for "Range" header value field.
+ */
+#define HTTP_RANGE_BYTES_VALUE_MAX_LEN          ( 27 )
```

Comment:
How was this calculated?

Reply:
HTTP_RANGE_FIELD_LEN + HTTP_HEADER_FIELD_SEPARATOR_LEN +  \
HTTP_RANGE_BYTES_PREFIX_VALUE_LEN + EQUAL_CHARACTER_LEN + \
INT32_STRING_MAX_LEN + DASH_CHARACTER_LEN + INT32_STRING_MAX_LEN;

Reply:
but i'm not working on that anymore so i'll just remove it